### ISSUE TITLE
Account signup user name validation issues

### ISSUE THREAD

**Issue by therealjeffg** [fission-suite/auth-lobby#82]
We're not catching some invalid username on signup. This was caught investigating #81 

Usernames like this:

* `jeff--`

...get an error like 

`Sorry, jeff-- is not a valid username. You can use letters, numbers and hyphens in between.`

After some testing, it looks like the code involved doesn't catch the '_' and '@' characters, leading to the creation of an invalid  personal address for the account.

**Comment by expede** [comment]
Yes, dashes or underscores at the beginning or end of names are not valid URLs, and thus fail server validation since it gets used as a subdomain. We should definitely make the message clearer about that.

**Comment by expede** [comment]
>  it looks like the code involved doesn't catch the '_' and '@' characters

Do you have an example of one with `@`? I wonder if the FE doesn't wait to check if it actually got created before trying to continue, because the server will reject those with a HTTP 422 (validation code [here](https://github.com/fission-suite/fission/blob/01dffb159aaa58afea83a771e9155db09311a9b6/fission-core/library/Fission/URL/Validation.hs#L19-L26)).

```haskell
> mkUsername "je-ff"
Right (Username "je-ff") -- Valid!

> mkUsername "hey@there"
Left Invalid -- Fails validation

> mkUsername "jeff--"
Left Invalid
```

**Comment by therealjeffg** [comment]
Yeah I can't repro the @ issue on mobile. Underscores are definitely a problem. 

I'll check macOS / Firefox again tomorrow 

![image](https://user-images.githubusercontent.com/104315/124226075-55515600-dabd-11eb-9f15-bb48d8c6eee7.png)

**Comment by icidasset** [comment]
Moving this to the webnative repo, since the username validation happens there.
